tests/integration/devfile/cmd_dev_test.go

```diff
@@ -0,0 +1,160 @@
+package devfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/redhat-developer/odo/pkg/util"
+	"github.com/redhat-developer/odo/pkg/watch"
+	"github.com/redhat-developer/odo/tests/helper"
+)
+
+var _ = Describe("odo dev command tests", func() {
+	var cmpName string
+	var commonVar helper.CommonVar
+
+	// This is run before every Spec (It)
+	var _ = BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		cmpName = helper.RandString(6)
+		helper.Chdir(commonVar.Context)
+	})
+
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	When("a component is bootstrapped and pushed", func() {
+		BeforeEach(func() {
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+			helper.Cmd("odo", "project", "set", commonVar.Project).ShouldPass()
+			helper.Cmd("odo", "init", "--name", cmpName, "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
+			//output := helper.Cmd("odo", "dev").ShouldPass().Out()
+			//Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
+		})
+		It("should show validation errors if the devfile is incorrect", func() {
+			session := helper.CmdRunner("odo", "dev")
+			defer session.Kill()
+			helper.WaitForOutputToContain("Waiting for something to change", 180, 10, session)
+			helper.ReplaceString(filepath.Join(commonVar.Context, "devfile.yaml"), "kind: build", "kind: run")
+			helper.WaitForOutputToContain(watch.PushErrorString, 180, 10, session)
+		})
+		It("should use the index information from previous push operation", func() {
+			// Create a new file A
+			fileAPath, fileAText := helper.CreateSimpleFile(commonVar.Context, "my-file-", ".txt")
+
+			// watch that project
+			session := helper.CmdRunner("odo", "dev")
+			defer session.Kill()
+
+			helper.WaitForOutputToContain("Waiting for something to change", 180, 10, session)
+
+			// Change some other file B
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "App started", "App is super started")
+			helper.WaitForOutputToContain("Executing devrun command", 180, 10, session)
+
+			podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+
+			// File should exist, and its content should match what we initially set it to
```

Comment:
Why? Shouldn't it match with the updated text?

Reply:
I copy-pasted the tests for `odo watch` command. But looking at the function, I think the content of file A is not changed during this test. So this comment is not incorrect in that sense.